flink-kubernetes-operator-autoscaler/src/main/java/org/apache/flink/kubernetes/operator/autoscaler/ScalingMetricCollector.java

```diff
@@ -223,16 +220,26 @@ private void updateKafkaSourceMaxParallelisms(
      * Given a map of collected Flink vertex metrics we compute the scaling metrics for each job
      * vertex.
      *
-     * @param collectedMetrics Collected metrics for all job vertices.
+     * @param flinkMetrics Collected metrics for all job vertices.
      * @return Computed scaling metrics for all job vertices.
      */
     private CollectedMetrics convertToScalingMetrics(
             ResourceID resourceID,
-            Map<JobVertexID, Map<FlinkMetric, AggregatedMetric>> collectedMetrics,
+            Map<JobVertexID, Map<FlinkMetric, AggregatedMetric>> flinkMetrics,
             JobTopology jobTopology,
             Configuration conf) {
 
         var out = new HashMap<JobVertexID, Map<ScalingMetric, Double>>();
+
+        var finishedVertices = jobTopology.getFinishedVertices();
+        Map<JobVertexID, Map<FlinkMetric, AggregatedMetric>> collectedMetrics;
+        if (finishedVertices.isEmpty()) {
+            collectedMetrics = flinkMetrics;
+        } else {
+            collectedMetrics = new HashMap<>(flinkMetrics);
+            finishedVertices.forEach(v -> collectedMetrics.put(v, FlinkMetric.FINISHED_METRICS));
+        }
```

Comment:
```suggestion
        if (!finishedVertices.isEmpty()) {
            collectedMetrics = new HashMap<>(collectedMetrics);
            finishedVertices.forEach(v -> collectedMetrics.put(v, FlinkMetric.FINISHED_METRICS));
        }
```

I would prefer this over adding an extra variable because it is easier to read for me.